src/data/transcripts/testing.md

```diff
@@ -0,0 +1,188 @@
+
+## Generating test cases with code
+
+Unison's [base library](https://github.com/unisonweb/base) contains powerful utility functions for generating test cases with Unison code, which lets your tests cover a lot more cases than if you are writing test cases manually like we've done so far. (This style of testing library is often called a _property-based testing library_.)
+
+The property-based testing support in Unison relies on [an ability](/docs/abilities) called `Gen` (short for "generator"). If you haven't read about [abilities](/docs/abilities) yet, we suggest taking a detour to do so before continuing.
+
+For instance, a `'{Gen} Nat` is a computation that, when run, produces `Nat` values. You can sample from a `'{Gen} a` multiple times to produce different values. Importantly, these are not _random_ values, the sequence of values generated is entirely deterministic:
+
+```unison
+> sample 100 (natIn 0 10)
+
+test> myTest = runs 100 '(expect (!(natIn 0 10) < 10))
+```
+
+When developing generators to use for testing, you'll often put those generators in a watch expression like this to make sure you understand what they are generating. Generators denote a set of values, and as the above shows, it is possible to exhaustively enumerate that set, at which point `sample` will stop short. Above we asked it to generate 100 natural numbers in the range 0 to 10, but there's only 10 unique values, so it stops after that.
+
+### Combining generators
+
+Where things get interesting is when _combining_ generators. There are a few ways of doing that. For a `'{Gen} a`, you can use the `!` operator to sample from it, and you can sample from multiple generators to build up a more complex generator. Let's have a look at an interesting example, which highlights something important about these generators:
+
+```unison
+> sample 10 'let
+  n = !(natIn 0 10) + 100
+  m = !(natIn 0 100)
+  (n, m)
+```
+
+> 🤓 __Syntax note:__ The `'let` is a common idiom to introduce a delayed computation which is a block. The precedence is such that `sample 10 'let ...` is parsed like `sample 10 ('let ...)`.
+
+As we can see, `Gen` does _fair_ or _breadth first_ sampling from both of the generators involved, rather than exhausting one before moving on to the next. (In this sense, it differs from the usual [`Choose` ability](/docs/abilities/#choose-for-expressing-nondeterminism) for nondeterminism which does depth-first enumeration.)
+
+Doing a breadth-first enumeration is the right move because as we build up more complex generators, where the space of possibilities is often so huge that it's only possible to sample a tiny fraction of it.
+
+There are two other ways of combining generators. One is `pick`, which fairly samples from multiple generators in a breadth first manner:
+
+```Haskell
```

Comment:
should be unison, not Haskell

Reply:
sometimes we say `haskell` just so that it will get highlighted, but probably it was a mistake